src/platform/datapath_winuser.c

```diff
@@ -1048,6 +1053,105 @@ CxPlatDataPathIsPaddingPreferred(
     return !!(Datapath->Features & CXPLAT_DATAPATH_FEATURE_SEND_SEGMENTATION);
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Success_(QUIC_SUCCEEDED(return))
+QUIC_STATUS
+CxPlatDataPathGetGatewayAddresses(
+    _In_ CXPLAT_DATAPATH* Datapath,
+    _Out_ QUIC_ADDR** GatewayAddresses,
+    _Out_ uint32_t* GatewayAddressesCount
+    )
+{
+    const ULONG Flags =
+        GAA_FLAG_INCLUDE_GATEWAYS |
+        GAA_FLAG_INCLUDE_ALL_INTERFACES |
+        GAA_FLAG_SKIP_DNS_SERVER |
+        GAA_FLAG_SKIP_MULTICAST;
+
+    UNREFERENCED_PARAMETER(Datapath);
+
+    ULONG AdapterAddressesSize = 0;
+    PIP_ADAPTER_ADDRESSES AdapterAddresses = NULL;
+    uint32_t Index = 0;
+
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+    ULONG Error;
+    do {
+        Error =
+            GetAdaptersAddresses(
+                AF_UNSPEC,
+                Flags,
+                NULL,
+                AdapterAddresses,
+                &AdapterAddressesSize);
+        if (Error == ERROR_BUFFER_OVERFLOW) {
+            free(AdapterAddresses);
```

Comment:
CXPLAT_FREE(AdapterAddresses, QUIC_POOL_DATAPATH_ADDRESSES);

Reply:
Also, in CXPLAT_FREE, we don't actually allow passing in NULL, as it dereferences the value to get the tag. So you need to protect this with an if not NULL.

Reply:
That's unfortunate. Are you sure we don't any other callers that assumed `CXPLAT_FREE(nullptr)` was acceptable in the code base? `free(nullptr)` and `delete nullptr` have always been acceptable.

Reply:
We can fixup CXPLAT_FREE to make that work. Its just currently in debug mode it does not work.

Reply:
Also, we should verify that ExFreePoolWithTag also has that same contract. I think it does, but the docs don't say.